### PR TITLE
Feat: greyscale preprocessor

### PR DESCRIPTION
**Pull request by SLAPaper**
Implemented #1427

### CODE REVIEW

File: scripts/processor.py [hunk 1]
```diff
@@ -91,6 +91,13 @@ def invert(img, res=512, **kwargs):
     return 255 - HWC3(img), True
 
 
+def greyscale(img, res=512, **kwargs):
+    """convert input image to greyscale
+    to be use with https://huggingface.co/ioclab/control_v1p_sd15_brightness
+    """
+    return cv2.cvtColor(HWC3(img), cv2.COLOR_RGB2GRAY, dstCn=3), True
```
Comment (thomaseding) [inline scripts/processor.py:98]: This should probably be
```
def greyscale(img, res=512, **kwargs):
    """convert input image to greyscale
    to be use with https://huggingface.co/ioclab/control_v1p_sd15_brightness
    """
    img, remove_pad = resize_image_with_pad(img, res)
    result = cv2.cvtColor(HWC3(img), cv2.COLOR_RGB2GRAY, dstCn=3)
    remove_pad(result), True
```

File: scripts/global_state.py [hunk 2]
```diff
@@ -83,6 +83,7 @@ def unified_preprocessor(preprocessor_name: str, *args, **kwargs):
     "shuffle": shuffle,
     "tile_resample": tile_resample,
     "invert": invert,
+    "greyscale": greyscale,
```
Comment (thomaseding) [inline scripts/global_state.py:86]: Add an entry to `preprocessor_aliases` to map `grayscale` alternative spelling

Comment (thomaseding) [reply scripts/global_state.py:86]: You can add an entry to `preprocessor_sliders_config`
```
    "greyscale": [
        {
            "name": flag_preprocessor_resolution,
            "value": 512,
            "min": 64,
            "max": 2048,
        }
    ],
```

and to `preprocessor_filters`
```
    "Greyscale": "greyscale",
```

**Comment by lllyasviel** [comment]
A bit late but I will say that this is going to be merged/implemented soon, together with SAI's recolor

**Pull request closed without merging**